doc/releases/changelog-dev.md

```diff
@@ -4,6 +4,9 @@
 
 <h3>New features since last release</h3>
 
+* A new `PennyLaneUserWarning` has been added to `exceptions` to warn users about special PennyLane tricks that might confuse them.
+  [(#7496)](https://github.com/PennyLaneAI/pennylane/pull/7496)
+
```

Comment:
I would probably consider this an improvement.

Reply:
Moved!